enterprise/views.py

```diff
@@ -552,6 +584,41 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
         return render_to_response('enterprise_course_enrollment_page.html', context_data, request=request)
 
     @method_decorator(enterprise_login_required)
+    def post(self, request, enterprise_uuid, course_id):
+        """
+        Process a submitted track selection form for the enterprise.
+        """
+        enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
+
+        selected_mode = request.POST.get('course_mode')
+
+        matching_modes = [mode for mode in modes if mode['mode'] == selected_mode]
+
+        if not matching_modes:
+            return self.get_enterprise_course_enrollment_page(request, enterprise_customer, course, modes)
+
+        if matching_modes[0].get('premium', False):
```

Comment:
No. BTW, Sylvia documented all the enrollment modes for DOC-3341 and the reference for that is here: http://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/glossary.html#enrollment-track-g

However, in this PR, @haikuginger is using `premium` to mean "any non-audit mode" - so it refers to `verified`, `professional`, and `professional-no-id`.